### PR TITLE
Default posture queries

### LINKED ISSUE #274: implement default posture callbacks
* make Posture Query type enum

### PR DESCRIPTION
**Pull request by ekoby**
[fixes #274]

### CODE REVIEW

File: library/posture.c [hunk 1]
```diff
@@ -254,8 +289,9 @@ void ziti_send_posture_data(ziti_context ztx) {
         if (ztx->opts->pq_domain_cb != NULL) {
             ztx->opts->pq_domain_cb(ztx, domainInfo->query->id, ziti_pr_handle_domain);
         } else {
-            ZITI_LOG(DEBUG, "%s cb not set requested for: service %s, policy: %s, check: %s", PC_DOMAIN_TYPE,
+            ZITI_LOG(VERBOSE, "using default %s cb for: service %s, policy: %s, check: %s", PC_DOMAIN_TYPE,
```
Comment (dovholuknf) [inline library/posture.c:292]: why this one to verbose when the others are left at debug?

Comment (ekoby) [reply library/posture.c:292]: i'll change all to verbose

File: library/posture.c [hunk 2]
```diff
@@ -545,3 +585,273 @@ static void ziti_pr_handle_process(ziti_context ztx, char *id, char *path, bool
...
+    switch (os_info.wProductType) {
+        case 1: os = "windows";
+            break;
+        case 2:
+        case 3: os = "windowsserver";
+            break;
+        default:
+            os = "<unknown windows type>";
+    }
+    char winver[16];
```
Comment (dovholuknf) [inline library/posture.c:613]: while this definitely seems like enough why not just make it bigger just because... it's short lived anyway. i worry :)

Comment (ekoby) [reply library/posture.c:613]: 32?

Comment (ekoby) [reply library/posture.c:613]: it won't crash, even if  the buffer is too short

File: library/posture.c [hunk 3]
```diff
@@ -545,3 +585,273 @@ static void ziti_pr_handle_process(ziti_context ztx, char *id, char *path, bool
...
+
+static void default_pq_domain(ziti_context ztx, const char* id, ziti_pr_domain_cb cb) {
+#if _WIN32
+    uint32_t status;
+    LPWSTR buf;
+    NetGetJoinInformation(NULL, &buf, &status);
+    char domain[256];
+    sprintf_s(domain, sizeof(domain), "%ls", buf);
+    cb(ztx, id, domain);
+    NetApiBufferFree(buf);
```
Comment (dovholuknf) [inline library/posture.c:661]: doesn't domain need to be free'd too?

Comment (ekoby) [reply library/posture.c:661]: no, domain is on stack

File: library/posture.c (hunk 2 above)
Comment (dovholuknf) [reply library/posture.c:613]: i just worry that it'll be a hard to track down bug is all. each of those fields "can be" up to like 10 characters long as a string, right? if MS makes them crazy long some day on purpose or by mistake it'd be a problem. 32 seems like it'd be enough but why not just go ham and use 64?

File: library/posture.c [hunk 4]
```diff
@@ -545,3 +585,273 @@ static void ziti_pr_handle_process(ziti_context ztx, char *id, char *path, bool
...
+                }
+                free(ex.ptr);
+            }
+        }
+    }
+
+#else
+    uv_utsname_t uname;
+    uv_os_uname(&uname);
+    ZITI_LOG(WARN, "not implemented on %s", uname.sysname);
```
Comment (dovholuknf) [inline library/posture.c:800]: better log message maybe? "cannot check for process running on %s. not implemented" ?

File: library/posture.c [hunk 5]
```diff
@@ -545,3 +585,273 @@ static void ziti_pr_handle_process(ziti_context ztx, char *id, char *path, bool
...
+
+    return rc;
+}
+
+static bool check_running(uv_loop_t *loop, const char *path) {
+    bool result = false;
+#if _WIN32
+    HANDLE sh = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
+    if (sh == NULL) {
+        ZITI_LOG(ERROR, "failed: %d", GetLastError());
```
Comment (dovholuknf) [inline library/posture.c:754]: why is the rest of the block allowed to continue if this fails? will line 763 still succeed if if `sh` is NULL?

File: library/posture.c (hunk 4 above)
Comment (ekoby) [reply library/posture.c:800]: log message shows function name

**Review by dovholuknf — APPROVED** [review]

File: library/posture.c [hunk 6]
```diff
@@ -795,6 +797,21 @@ static bool check_running(uv_loop_t *loop, const char *path) {
...
+#elif __APPLE__ && __MACH__
+    int n_pids = proc_listallpids(NULL, 0);
+    unsigned long pids_sz = sizeof(pid_t) * (unsigned long)n_pids;
+    pid_t * pids = calloc(1, pids_sz);
+    proc_listallpids(pids, (int)pids_sz);
+    char proc_path[PROC_PIDPATHINFO_MAXSIZE];
+    for (int i=0; i < n_pids; i++) {
+        if (pids[i] == 0) continue;
+        proc_pidpath(pids[i], proc_path, sizeof(proc_path)); // returns strlen(proc_path)
+        if (strncasecmp(proc_path, path, sizeof(proc_path)) == 0) {
```
Comment (ekoby) [inline library/posture.c:809]: I don't believe it should be case-insensitive

Comment (scareything) [reply library/posture.c:809]: Why not? the default file systems on macOS are case-insensitive:
```
$ uname -s
Darwin
$ touch Foo
$ touch foo
$ ls ?oo
Foo
```

Comment (ekoby) [reply library/posture.c:809]: :exploding_head: I did not know that

**Pull request merged**